routes/articles.js

```diff
@@ -0,0 +1,90 @@
+/* jshint esversion:6 */
+const express = require('express');
+const router = express.Router();
+const hbs = require('express-handlebars');
+
+const Articles = require('../db/articles');
+const articles = new Articles();
+
+const validReq = { "success" : true };
+const invalidReq = { "success" : false };
+
+router.route('/')
+  .get((req, res) => {
+    res.render('index', { 
+      articles : {
+        list : articles.listAll() 
+      }
+    });
+  });
+
+router.route('/new')
+  .get((req, res) => {
+    res.render('index', { 
+      articles : {
+        new : true 
+      }
+    });
+  })
+
+  .post((req, res) => {
+    if (articles.create(req.body)) return res.redirect('/articles')/*res.redirect('/articles')*/;
+    else return res.redirect('/articles/new')/*res.redirect('/articles/new')*/;
+  });
+
+router.route('/:title')
+  .get((req, res) => {
+    let title = req.params.title;
+    console.log(title);
+    let targetItem = articles.find(title);
+    console.log(targetItem);
+
+    if (targetItem) { 
+      return res.render('index', { 
+        articles : {
+          item : targetItem 
+        }
+    });
+    // I bet this will break...I think I'm wrong
+    } else {
+      return res.redirect(`/articles`);
+    }
+  });
+
+router.route('/:title/edit')
+  .get((req, res) => {
+    let title = req.params.title;
+    let targetItem = articles.find(title);
+    console.log(targetItem);
+
+    if (targetItem) { 
+      return res.render('index', { 
+        articles : {
+          edit: true,
+          title : title,
+          item : targetItem 
+        }
+    });
+    // I bet this will break...I think I'm wrong
+    } else {
```

Comment:
Your `title` and `targetItem` in the end are references to the same thing, so I `targetItem` was not found you would not be able to redirect to the `title` view